### ISSUE TITLE
MRM_hat becomes NaN when fine-tuning a pretrained model L-Mask-offline

### ISSUE THREAD

**Issue by abeer777313** [Audio-WestlakeU/CleanMel#4]
hello
I'm encountering a problem when fine-tuning a pretrained CleanMel L-mask-Offline  model on a new dataset. Specifically, during training, the MRM_hat values become NaN, and consequently the loss also becomes NaN.

I'm loading a pretrained model checkpoint and fine-tuning it on a new dataset.
The training precision is set to FP16 (precision=16).
I modified the arch/cleanmel.py code to ensure the optimizer uses Adam with eps=1e-4, as recommended for mixed-precision training.
I've also tried increasing   value  in den in mask from 1e-10 to 1e-5, but the issue persists.
Due to memory constraints, I'm training with a batch size of 1.
Any suggestions on how to stabilize training and prevent NaNs in the mask outputs would be appreciated.

**Comment by SaoYear** [comment]
Hi, 

Thank you for your interest in our work.

As for your problem, I would provide some experiences:
1. The training of CleanMel requests large batch size, in your case, I would suggest you to train a small model first.
2. There could be several reasons that lead to NaN value in the predicted mask. Could you please provide the training curve or checkout if there exists any strange spikes? This could happen in the beginning of the training (e.g., first 312,500 steps)
3. To stablize the training, you could add a tempature value to the output mask **before Sigmoid**, which could avoid gradient  explosion from my experience. E.g.,
```
output = output / t
output = Sigmoid(output)
# t decays during first 3 epochs
```
4. For training precision, I never trained the model in FP16. In default precision,  AdamW performed best for Mamba-based network, but I believe the Adam would not make the training process collapsed.

**Comment by SaoYear** [comment]
Feel free to re-open this issue if you have further question : )
